sklearn/linear_model/_sag_fast.pyx.tp

```diff
@@ -23,10 +23,6 @@ License: BSD 3 clause
 dtypes = [('64', 'double', 'np.float64'),
           ('32', 'float', 'np.float32')]
 
-def get_dispatch(dtypes):
-    for name, c_type, np_type in dtypes:
-        yield name, c_type, np_type
-
```

Comment:
Nice catch! This can indeed also be removed here for SAG and SGD.

The original scope of this PR was to port `WeightVector` to 32bits using Tempita, but, to me, we can extend it to have other change such as those welcomed simplification.

In this case, the PR's goal/title/description could be changed for something like "MAINT Improve source generation using `Tempita`"

Reply:
FYI, I think that this is just for some historical reason. We borrow the tempita way of doing from pandas that uses the following: https://github.com/pandas-dev/pandas/blob/059c8bac51e47d6eaaa3e36d6a293a22312925e6/pandas/_libs/algos_common_helper.pxi.in#L39-L58

At the time of introducing it (2016), it might have exist some limitation between tempita and cython

Reply:
I also see the same pattern in scipy: https://github.com/scipy/scipy/blob/701ffcc8a6f04509d115aac5e5681c538b5265a2/scipy/sparse/_csparsetools.pyx.in

Going a bit further, I found the following: https://pyrocore.readthedocs.io/en/latest/tempita.html#for
I am not sure if we got trapped in the issue of nested tuple where the dispatcher was a way to bypass it.